plugins/modules/purefa_pgsched.py

```diff
@@ -305,31 +320,76 @@ def update_schedule(module, array, snap_time, repl_time):
             "per_day": per_day,
             "all_for": all_for,
         }
-        module.warn("current {0}; new: {1}".format(current_snap, new_snap))
         if current_snap != new_snap:
             changed = True
             if not module.check_mode:
                 try:
-                    array.set_pgroup(
-                        module.params["name"], snap_enabled=module.params["enabled"]
-                    )
+                    if LooseVersion(CONTEXT_API_VERSION) <= LooseVersion(api_version):
+                        array.patch_protection_groups(
+                            names=[module.params["name"]],
+                            context_names=[module.params["context"]],
+                            protection_group=ProtectionGroup(
+                                snapshot_schedule=SnapshotSchedule(
+                                    enabled=new_snap["snap_enabled"],
+                                    frequency=new_snap["snap_frequency"],
+                                )
+                            ),
+                        )
+                    else:
+                        array.patch_protection_groups(
+                            names=[module.params["name"]],
+                            protection_group=ProtectionGroup(
+                                snapshot_schedule=SnapshotSchedule(
+                                    enabled=new_snap["snap_enabled"],
+                                    frequency=new_snap["snap_frequency"],
```

Comment:
possibly bug in API, whatever value we pass, snap frequency always set to "3600000", pls cross check on the same.

Reply:
Please recheck this with the latest push. I can't find a direct issue, but it may be an artifact of another issue that has been fixed.